packages/picasso-forms/src/RichTextEditor/hooks/use-enforce-highlight-autofill.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useMemo, useState } from 'react'
+
+import { useFormConfig } from '../../FormConfig'
+
+/**
+ * This hook keeps the highlighted state for Rich Text Editor form fields after
+ * first `onChange` callback which is triggered by editor itself when initial value is
+ * provided. Starting from second `onChange` or first `onFocus` invocation (which are
+ * triggered by user) the field is no longer highlighted.
+ *
+ * Editor triggers the very first onChange callback by itself only when initial value
+ * is provided in order to propagate the value to wrapping form.
+ */
+
+const TRACKED_EVENTS_LIMIT = 3
+
+export const useEnforceHighlightAutofill = () => {
+  const { highlightAutofill } = useFormConfig()
+  const [timesChangeOrFocusTriggered, setTimesChangeOrFocusTriggered] =
+    useState(0)
+
+  const enforceHighlightAutofill = useMemo(() => {
+    if (!highlightAutofill) {
+      return false
+    }
+
+    return timesChangeOrFocusTriggered < 2
+  }, [highlightAutofill, timesChangeOrFocusTriggered])
+
+  const registerChangeOrFocus = useCallback(() => {
+    if (!highlightAutofill) {
+      return
+    }
+
+    if (timesChangeOrFocusTriggered < TRACKED_EVENTS_LIMIT) {
+      setTimesChangeOrFocusTriggered(timesChangeOrFocusTriggered + 1)
```

Comment:
```suggestion
      setTimesChangeOrFocusTriggered(times => times + 1)
```

to rely on the current `times`

Reply:
@dmaklygin Can you please elaborate on why it is needed if we have a `timesChangeOrFocusTriggered` as a dependency of `useCallback()`, so it is always fresh?